### PR TITLE
Add CSAT

### PR DESCRIPTION
**Pull request by HardeepAsrani**
### All Submissions:

* [x] Have you followed the [Optimole Contributing guideline](https://github.com/Codeinwp/optimole-wp/blob/master/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change? 
 

### Changes proposed in this Pull Request:

Screenshot 2023-06-26 at 9 50 47 AM

This adds CSAT notice to Optimole Dashboard. The popup will be displayed when the following conditions are met:

- The user has completed the setup process (API connection).
- The user has visited any of the settings tabs three times.
- The user has optimized images.
- It is user's first website with Optimole, so if you connect the same account to another website, it shouldn't appear.
- Scores should be recorded regardless of whether the user leaves a comment.
- If you add a score leave the page (or refresh), the API request will be made when the page finishes reloading or the user comes back to the Optimole page.

Closes Codeinwp/optimole-service#893

### How to test the changes in this Pull Request:

1. Make sure all the conditions that are specified in the above section are met, and the CSAT popover appears only when the conditions are met.
2. Request is made properly and the CSAT doesn't appear when it's not supposed to appear, ie once it has been dismissed by the user or a request has been made already.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by abaicus — COMMENTED** [review]
Things seem fine 👍🏻 

Just one small suggestion I added. Let me know what you think! Thanks!

File: assets/src/parts/connected/CSAT.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+	const {
+		hasImages,
+		isFirstSite
+	} = useSelect( select => {
+		const { getUserData } = select( 'optimole' );
+
+		return {
+			hasImages: 0 < getUserData()?.images_number,
+			isFirstSite: 1 === getUserData()?.app_count
```
[the remark this answers is not in the document]
Comment (HardeepAsrani) [reply assets/src/parts/connected/CSAT.js:43]: > If a user connects two websites, he'll never see this again.

What do you mean? We don't want to show this to a user for the second time, so hence only targeting fresh sites: https://github.com/Codeinwp/optimole-service/issues/893#issuecomment-1593550311

[the remark this answers is not in the document]
Comment (abaicus) [reply assets/src/parts/connected/CSAT.js:43]: Indeed, I missed that comment. Then it should be fine 😄

**Review by abaicus — APPROVED** [review]

**Comment by rodica-andronache** [comment]
@HardeepAsrani 

- [ ] I have the same account connected on multiple sites https://vertis.d.pr/i/RODFic and the popup appears on each of them https://vertis.d.pr/i/ShDf5c https://vertis.d.pr/i/ByaRfC

- [ ] the `What info we collect?` link does not work https://vertis.d.pr/i/2RKzs5

- [ ] The popup layout on mobile https://vertis.d.pr/i/I3EdSE

- [ ] there is no Thank you message after the popup is submitted, is that ok? In the initial issue, that is mentioned

**Comment by HardeepAsrani** [comment]
Should work fine now. For the notice, it's a tooltip so you will have to hover it.

**Comment by rodica-andronache** [comment]
@HardeepAsrani thank you for the fixes!
As I've mentioned on slack, I've tried multiple times to test this and sometimes, the notice appears just after connecting the API key, before I go into Settings for three times. The notice does disappear if I remove `optimole_settings_visits` from localStorage. If that's normal, we can leave it like that, I just wanted to mention it.

Apart from that, the notice doesn't disappear even if I click on the close button, or the submit button, it appears again and again https://vertis.d.pr/v/ODR1nG. Let me know if there's something I'm doing wrong here.

And a small thing, on mobile, maybe the layout could be different for the skip/submit button as they are a bit squished. Also, the "What info we collect?" tooltip can't be used on mobile, as you can't hover it.

**Comment by HardeepAsrani** [comment]
@rodica-andronache Dismiss issue should be fixed now. For the mobile part, we can't do much as core component doesn't support it, we can let it be for now.

**Comment by rodica-andronache** [comment]
@HardeepAsrani it's working well now, thank you!

**Pull request merged**